### PR TITLE
Add ZCore

### PR DESCRIPTION
**Pull request by apollomatheus**
Hello, 

I did some changes to add ZCore.

**Blockbook**
https://blockbook.zcore.cash

**More information:**
https://zcore.cash
https://zcore.info
https://github.com/zcore-coin/zcore-2.0

Chainparams:
https://github.com/zcore-coin/zcore-2.0/blob/master/src/chainparams.cpp

### CODE REVIEW

File: common/defs/support.json [hunk 1]
```diff
@@ -1525,6 +1527,7 @@
       "bitcoin:XSN": "2.0.11",
       "bitcoin:XZC": "2.0.7",
       "bitcoin:ZCL": "2.0.11",
+      "bitcoin:ZCR": "2.0.11",
```
Comment (prusnak) [inline common/defs/support.json]: Should be `2.1.7`

File: common/defs/support.json [hunk 2]
```diff
@@ -160,6 +161,7 @@
       "bitcoin:XSN": "1.8.0",
       "bitcoin:XZC": "1.6.2",
       "bitcoin:ZCL": "1.8.0",
+      "bitcoin:ZCR": "1.8.0",
```
Comment (prusnak) [inline common/defs/support.json]: Should be `1.8.4`

**Comment by zcore-coin** [comment]
Hi @prusnak 
Can you include it in Trezor Beta?
If not possible we will develop our own or try it at Magnum Wallet.
Let us also implement Trezor as authentication on our platform: https://central.zcore.cash/mn-coins

**Comment by zcore-coin** [comment]
> Which wallet do you plan to integrate Trezor to?



> Which wallet do you plan to integrate Trezor to?

Hi @prusnak
Can you include it in Trezor Beta?
If not possible we will develop our own or try it at Magnum Wallet.
Let us also implement Trezor as authentication on our platform: https://central.zcore.cash/mn-coins

**Comment by prusnak** [comment]
Fix issues I posted above and discuss with Magnum whether they will add Trezor support for your coin.

**Comment by apollomatheus** [comment]
All issues fixed, please review.

**Comment by prusnak** [comment]
@apollomatheus no they are not, the tests are still red. Run `make templates` and commit the result

**Comment by apollomatheus** [comment]
@prusnak I run `make templates`, but it looks that it removed eth tokens.

**Comment by prusnak** [comment]
The changes look OK now.

File: common/defs/bitcoin/zcore.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
+{
+  "coin_name": "ZCore",
+  "coin_shortcut": "ZCR",
+  "coin_label": "ZCore",
+  "website": "https://zcore.cash",
+  "github": "https://github.com/zcore-coin/zcore-2.0",
+  "maintainer": "Erick Costa <erick@zcore.cash>",
+  "curve_name": "secp256k1",
+  "address_type": 142,
+  "address_type_p2sh": 0,
```
Comment (prusnak) [inline common/defs/bitcoin/zcore.json]: Are you sure this is the correct address_type?

We can't accept a coin which has address version that collides with Bitcoin (that is addresses start with `1` or `3`).

Comment (apollomatheus) [reply common/defs/bitcoin/zcore.json]: It's correct, 142 starts with `z` only.
I've tested the firmware and it generates addresses with `z`.

https://en.bitcoin.it/wiki/List_of_address_prefixes

Comment (prusnak) [reply common/defs/bitcoin/zcore.json]: I am talking about `0` for address_type_p2sh.
It is indeed 0 here:
https://github.com/zcore-coin/zcore-2.0/blob/master/src/chainparams.cpp#L176
but that is unacceptable value. It means that multisig addresses of ZCore will start with `1` resembling the Bitcoin addresses.

Please change this value in zcore and here and then we will merge.

File: common/defs/bitcoin/zcore.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  "minfee_kb": 1000,
+  "signed_message_header": "ZCore Genesis Block mined by Mosqueiro",
+  "hash_genesis_block": "695b79c8c234b45b2eeb4722f33373e471c9b686ff78efeb39da95f824a9f81b",
+  "xprv_magic": 78791432,
+  "xpub_magic": 78792518,
+  "xpub_magic_segwit_p2sh": null,
+  "xpub_magic_segwit_native": null,
+  "bech32_prefix": null,
+  "cashaddr_prefix": null,
+  "slip44": 177,
```
Comment (prusnak) [inline common/defs/bitcoin/zcore.json]: This should be 428

**Comment by matejcik** [comment]
@prusnak should we add a check for conflict of `address_type_p2sh` with Bitcoin's (or other coins') `address_type`s?

**Comment by zcore-coin** [comment]
@prusnak please add our Electrum Wallet :
https://github.com/zcore-coin/electrum-wallet
Thank you!

**Comment by prusnak** [comment]
@zcore-coin Please send a pull request

**Comment by zcore-dev** [comment]
> @zcore-coin Please send a pull request to `common/defs/wallets.json` file

@prusnak done https://github.com/trezor/trezor-firmware/pull/764

**Pull request merged**